src/Symfony/Component/JsonPath/JsonCrawler.php

```diff
@@ -310,7 +320,32 @@ private function evaluateFilter(string $expr, mixed $value): array
 
     private function evaluateFilterExpression(string $expr, mixed $context): bool
     {
-        $expr = trim($expr);
+        $expr = JsonPathUtils::normalizeWhitespace($expr);
+
+        // remove outer parentheses if they wrap the entire expression
+        if (str_starts_with($expr, '(') && str_ends_with($expr, ')')) {
+            $depth = 0;
+            $isWrapped = true;
+            for ($i = 0; $i < strlen($expr); $i++) {
+                if ($expr[$i] === '(') {
+                    $depth++;
+                } elseif ($expr[$i] === ')') {
+                    $depth--;
+                    if ($depth === 0 && $i < strlen($expr) - 1) {
+                        $isWrapped = false;
+                        break;
+                    }
+                }
+            }
```

Comment:
/cc @fabpot we have an issue with fabbot:

```suggestion
        if (str_starts_with($expr, '(') && str_ends_with($expr, ')')) {
            $depth = 0;
            $isWrapped = true;
            for ($i = 0; $i < strlen($expr); ++$i) {
                if ('(' === $expr[$i]) {
                    ++$depth;
                } elseif (')' === $expr[$i]) {
                    --$depth;
                    if (0 === $depth && $i < \strlen($expr) - 1) {
                        $isWrapped = false;
                        break;
                    }
                }
            }
```